First phase

JoseFlores-baz opened the pull request:
Hola  Luis, por favor te pido realices la revisión de mi primer entregable, contiene los siguientes cambios:

* Crear un cliente con Retrofit2 y  GsonAdapter para consumir:
    * https://bitso.com/api_info#available-books
    * https://bitso.com/api_info#ticker
    * https://bitso.com/api_info#order-book

* Para las imágenes que representen las monedas usé la api https://cryptoflash-icons-api.herokuapp.com/ y la librería Picasso https://square.github.io/picasso/ para cargarlas.

* Usé MVVM y LiveData para poder diseñar la aplicación:
    * Implemente las capas recomendadas para aplicaciones Android(Data, Domain, UI)
    * Implementé una pantalla para mostrar la lista de las monedas. 
    * Implementé una pantalla para mostrar el detalle de la cripto divisa y las listas de ofertas y demandas.
    * En esta parte usé ListAdapter y ConstrainLayout, ya que necesitaba ejercitar su implementación.

* Implementé Navigation Component para navegar entre los fragmentos.

* Hice uso de las buenas prácticas que conozco( me faltó comentar algunas interfaces, espero hacerlo en la siguiente entrega)

Saludos, adjunto el video demo:

https://user-images.githubusercontent.com/116051350/199281537-a4b73922-881b-4c9b-9c7a-cb73123e1dff.mov

Commit:
data layer implementation progress

Commit:
repositories, use cases and data sources implementation progress, refactor names of models

Commit:
add gitignore

Commit:
create recycler view on main activity (item_layout, adapter and modify main_activity_xml )
implements viewModel and methods to invoke fetchBooks

Commit:
clean code and changes after mentoring 1

Commit:
move toDomain code to mappers

Commit:
resolve gitignore issue

Commit:
resolve gitignore issue

Commit:
resolve gitignore issue

Commit:
resolve gitignore issue

Commit:
Second screen design implementation progress

Commit:
add navigation fragment
add roboto fonts
refactor main activity
add functions in ViewUtils

JoseFlores-baz committed:
Delete file-system.probe

JoseFlores-baz committed:
Delete executionHistory.bin

JoseFlores-baz committed:
Delete executionHistory.lock

Commit:
edit git ignore

Commit:
Merge branch 'first_phase' of https://github.com/JoseFlores-baz/baz-android-course into first_phase

Commit:
delete unnecessary files

Commit:
delete unnecessary files .idea folder

Commit:
delete unnecessary files .idea folder 2

Commit:
first phase finised

Commit:
first phase finised

luismario-wize commented on app/src/main/java/com/example/bazandroidcourse/ui/utils/ViewUtils.kt:
```diff
@@ -0,0 +1,35 @@
+
+fun createURLImageById( tickerId:String, size:Int = 32):String {
+    // todo:send this to build field
+    return "https://cryptoflash-icons-api.herokuapp.com/$size/${tickerId}"
```
Pasar al flavor en gradle

JoseFlores-baz replied:
Listo!

luismario-wize commented on app/src/main/java/com/example/bazandroidcourse/data/utils/network/NetworkManagerImpl.kt:
```diff
@@ -0,0 +1,26 @@
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+class NetworkManagerImpl (val context: Context):NetworkManagerInterface{
```
Se puede convertir en un singleton ? 
Ejemplo:
`object NetworkManagerUtil { 
val context: Context? = null
override fun isOnline(): Boolean {
        var result = false
        context?.let {
        val connectivityManager =
            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkCapabilities = connectivityManager.activeNetwork ?: return false
        val actNw =
            connectivityManager.getNetworkCapabilities(networkCapabilities) ?: return false
        result = when {
            actNw.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) -> true
            actNw.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) -> true
            actNw.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET) -> true
            else -> false
        }
}
        return result
    }

`

JoseFlores-baz replied:
Listo, solo que hubo que agregar @SuppressLint("StaticFieldLeak") porque mostraba un warning

luismario-wize commented on app/src/main/java/com/example/bazandroidcourse/ui/viewmodel/BooksViewModel.kt:
```diff
@@ -0,0 +1,70 @@
+
+    private val getBooksUseCase = GetAllBooksFilteredUseCase(
+        BooksRepositoryImpl(
+            NetworkManagerImpl(context),
```
En lugar de pasarlo por parametro utiliza el singleton

JoseFlores-baz replied:
Listo!

Commit:
changes after first phase review:
-moves creation of viewModel dependencies creation to fragments.
-refactor NetworkManagerImpl.kt
-optimize imports.

Commit:
changes after merge
